framework/core/Utils.pm

```diff
@@ -257,6 +266,52 @@ sub check_vid {
     return {valid => 1, bid => $1, type => $2};
 }
 
+=pod
+  Utils::ensure_valid_bid(pid, bid)
+
+Ensure C<bid> represents a valid bug-id in project C<cid>, terminating with a
+helpful error message if not. A bug-id is valid for a project if the project
+exists and the bug-id both exists in the project and has not been deprecated.
+=cut
+sub ensure_valid_bid {
+    @_ == 2 or die $ARG_ERROR;
+    my $pid = shift;
+    my $bid = shift;
+    my $project = Project::create_project($pid);
+    my $project_dir = "$SCRIPT_DIR/projects/$pid";
+
+    if ( ! -e "${project_dir}" ) {
+        confess("Error: ${pid}-${bid} is a non-existent bug\n");
```

Comment:
This should probably be `${pid} is a non-existent project`

Reply:
done